tools/clang/lib/SPIRV/SpirvEmitter.h

```diff
@@ -1391,6 +1395,10 @@ class SpirvEmitter : public ASTConsumer {
   /// called from the entry point function.
   FunctionDecl *patchConstFunc;
 
+  /// Whether or not an Interlock execution mode has been added to the entry
+  /// function.
+  bool interlockModeAdded;
```

Comment:
Yes, in `beginInvocationInterlock`: https://github.com/microsoft/DirectXShaderCompiler/pull/5820/files#diff-6b86e3fcb3f49e0a10dd39cc10c7ec49d6c00bf11422c86743ef63a34eac6303R761.

This seems to be needed because if you `spvBuilder.addExecutionMode(entryFunction, executionMode)` multiple times, multiple duplicate `OpExecutionMode` instructions will be duplicated.

Reply:
Looks like Steven has a better way of doing this in https://github.com/microsoft/DirectXShaderCompiler/pull/5829/files#diff-6d199cfeca02410020b77a23e3bc3c92e9aab2ed24d8769bbf2fdffcb88bcbfaR923

Reply:
I think this would be better. As it can also handles execution mode conflict:
 it it's already set by another extension, we can say "something is wrong". As adding a new interlockModeAdded bit would require explicitely checking for it.

Reply:
I've updated the code to use Steven's approach.